tidy3d/components/validators.py

```diff
@@ -464,3 +466,19 @@ def validate_mode_plane_radius(mode_spec: ModeSpec, plane: Box, msg_prefix: str
             f"{msg_prefix} bend radius is smaller than half the mode plane size "
             "along the radial axis, which can produce wrong results."
         )
+
+
+def _warn_unsupported_traced_argument(name: str):
+    @pydantic.validator(name, always=True, allow_reuse=True)
+    def _warn_traced_arg(cls, val, values):
+        if isbox(val):
+            log.warning(
+                f"Field '{name}' of '{cls.__name__}' received an autograd tracer "
+                f"(a boxed value). Automatic differentiation through this field is "
```

Comment:
people might not know what a boxed value is? also might be good to mention that users can apply `getval` to the value to make it untraced.